src/widgets/panel/panel.tsx

```diff
@@ -103,14 +114,14 @@ export type PanelWidgetOptions = {
 const withUsage = createDocumentationMessageGenerator({ name: 'panel' });
 const suit = component('Panel');
 
-const renderer = ({
+const renderer = <TWidget extends UnknownWidgetFactory>({
   containerNode,
   bodyContainerNode,
   cssClasses,
   templates,
 }) => ({ options, hidden, collapsible, collapsed }) => {
   render(
-    <Panel
+    <Panel<TWidget>
```

Comment:
Wow—since when is this syntax supported? Wasn't aware of it!

Reply:
Yeah, Haroen introduced it to us in the stacked PR: https://github.com/algolia/instantsearch.js/pull/4558
and it's interesting!

Reply:
a long time already, TS 2.9: https://mariusschulz.com/blog/passing-generics-to-jsx-elements-in-typescript

Reply:
Damn, I wish I had known that sooner!